lib/saluki-config/src/refresher.rs

```diff
@@ -0,0 +1,206 @@
+use std::sync::Arc;
+
+use arc_swap::ArcSwap;
+use reqwest::ClientBuilder;
+use saluki_error::GenericError;
+use serde::{de::DeserializeOwned, Deserialize};
+use serde_json::Value;
+use tokio::time::{sleep, Duration};
+use tracing::{debug, error};
+
+use crate::{ConfigurationError, GenericConfiguration};
+
+const DEFAULT_AGENT_IPC_HOST: &str = "localhost";
+const DEFAULT_AUTH_TOKEN_FILE_PATH: &str = "/etc/datadog-agent/auth_token";
+const DEFAULT_REFRESH_INTERVAL_SECONDS: u64 = 15;
+
+/// Configuration for setting up `RefreshableConfiguration`.
+#[derive(Default, Deserialize)]
+pub struct RefresherConfiguration {
+    /// The location of the auth token used by the Datadog Agent.
+    ///
+    /// Defaults to `/etc/datadog-agent/auth_token`.
+    #[serde(default = "default_auth_token_file_path")]
+    auth_token_file_path: String,
+
+    /// The amount of time betweeen each request in seconds.
+    ///
+    /// Defaults to 15 seconds.
+    #[serde(
+        rename = "agent_config_refresh_internal_seconds",
```

Comment:
Sorry for the flip-flopping here, but now that I see how we're aligning more with the `agent_ipc` stuff in the Agent, I think we would actually want this to be `agent_ipc_config_refresh_interval`, which would align with how it's used by the `comp/core/configsync` component.